src/wp-includes/script-loader.php

```diff
@@ -1843,6 +1843,77 @@ function wp_just_in_time_script_localization() {
 	);
 }
 
+/**
+ * Prints a loader script if there is text/template registered script.
+ *
+ * @since 6.3.0
+ *
+ * When added to the DOM, this script converts any text/template scripts
+ * associated with a handle into type/javascript, and executes them.
+ */
+function wp_print_delayed_inline_script_loader() {
+	$wp_scripts = wp_scripts();
+
+	if ( $wp_scripts->has_delayed_inline_script() ) {
+		$output            = <<<JS
+(function () {
+  var nonce = document.currentScript.nonce;
+
+  /**
+   * Load event handler.
+   *
+   * @param {Event} event Event.
+   */
+  function onScriptLoad(event) {
+    var i, len, newScript, matches, scripts;
+    if (
+      !(
+        event.target instanceof HTMLScriptElement ||
+        event.target.async ||
+        event.target.defer ||
+        event.target.id
+      )
+    ) {
+      return;
+    }
+    matches = event.target.id.match(/^(.+)-js$/);
+    if (!matches) {
+      return;
+    }
+    scripts = document.querySelectorAll(
+      '[type="text/template"][data-wp-executes-after="' + matches[1] + '"]'
+    );
+    for (i = 0, len = scripts.length; i < len; i++) {
+      if (nonce && nonce !== scripts[i].nonce) {
+        console.error(
+          "CSP nonce check failed for after inline script. Execution aborted.",
+          scripts[i]
+        );
+        continue;
+      }
+      newScript = scripts[i].cloneNode(true);
+      newScript.type = "text/javascript";
+      scripts[i].parentNode.replaceChild(newScript, scripts[i]);
+    }
+  }
+  document.addEventListener("load", onScriptLoad, true);
+
+  window.addEventListener(
+    "load",
+    () => {
+      document.removeEventListener("load", onScriptLoad, true);
+    },
+    { once: true }
+  );
+})();
+JS;
+		$type_attr         = current_theme_supports( 'html5', 'script' ) ? '' : 'text/javascript';
+		$script_type_array = '' === $type_attr ? array() : array( 'type' => $type_attr );
+
+		wp_print_inline_script_tag( $output, array_merge( $script_type_array, array( 'id' => 'wp-executes-after-js' ) ) );
+	}
+}
```

Comment:
Imho support for "script after" for async and defer scripts doesn't make sense. The current implementation doesn't seem particularly good either. Seems better to remove this for now and eventually add it if/when the need for it is established. Also, there seem to be better ways to add such scripts that don't necessarily need to be handled by Script Loader. Example: the inline script can be outputted/printed at any time and run on `DOMContentLoaded`. 

See discussion on Trac starting from: https://core.trac.wordpress.org/ticket/12009#comment:112.